mlir/include/mlir/Dialect/DLTI/DLTI.h

```diff
@@ -33,6 +33,17 @@ DataLayoutSpecInterface getDataLayoutSpec(Operation *op);
 /// DataLayoutOpInterface, a method on ModuleOp, or an attribute implementing
 /// the interface, on `op` and else on `op`'s ancestors in turn.
 TargetSystemSpecInterface getTargetSystemSpec(Operation *op);
+
+/// Retrieve the first `DLTIQueryInterface`-implementing attribute that is
+/// attached to `op` or such an attr on as close as possible an ancestor. The
+/// op the attribute is attached to is returned as well.
+std::pair<DLTIQueryInterface, Operation *> getClosestQueryable(Operation *op);
```

Comment:
If someone were to want to reimplement `query` themselves, this (kind of) function would probably be useful. Having said that, a more natural way to extend the DLTI dialect is to just implement your own attributes. If they implement the `DLTIQueryInterface`, their attributes will automatically work with `dlti::query` - which is presumably the main way everyone would like to engage with target information.

I have pushed a new commit that no longer exposes this utility function (and removes two that were introduced in  https://github.com/llvm/llvm-project/pull/101561). If people would like to call utility functions like the above, they could always re-expose them.

Reply:
At least it could be a private method or something, doesn't matter. However, I would not remove this without further consultation. @ftynse @Dinistro @banach-space

Reply:
The commit I added today makes the `getClosestQueryable` method/function private.

The two functions I removed - `getDataLayoutSpec` and `getTargetSystemSpec` - are the two I introduced in https://github.com/llvm/llvm-project/pull/101561 so that `transform.dlti.query` could use them to implement the query logic. As that logic has now moved into `dlti::query()` and `transform.dlti.query` makes use of that function directly, there are no more users of these (very recently introduced) public methods/functions.

That is, there is no risk of breaking downstream users as they would have needed to start using these public methods in the last week or so.

Reply:
Ah, indeed! Then no worries, looks good, thanks!